### PR TITLE
testify: support testify in ddl/restart_test.go

### LINKED ISSUE #29109: migrate test-infra to testify for `ddl/restart_test.go`

### PR DESCRIPTION
**Pull request by wangggong**
Signed-off-by: wangggong <793160615@qq.com>

### What problem does this PR solve?

Issue Number: close #29109

Problem Summary:

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/restart_test.go [hunk 1]
```diff
@@ -104,6 +113,31 @@ LOOP:
...
+			break LOOP
+		}
+	}
+}
+
+func testCheckOwnerT(t *testing.T, d *ddl, expectedVal bool) {
+	require.Equal(t, expectedVal, d.isOwner())
+}
+
 func (s *testSchemaSuite) TestSchemaResume(c *C) {
```
Comment (hawkingrei) [inline ddl/restart_test.go:141]: Can you help this function to support testify?

Comment (hawkingrei) [reply ddl/restart_test.go:141]: Ditto ```TestSchemaResume```

Comment (wangggong) [reply ddl/restart_test.go]: OK, done

PTAL

Comment (wangggong) [reply ddl/restart_test.go]: NOTE that `getDDLSchemaVer` is deleted 'cause linter will warning that the method is not used.

**Pull request closed without merging**